### PR TITLE
Model configuration output in HDF

### PR DESCRIPTION
**Pull request by andrewfullard**
## Description

Unit test added for a `to_hdf` method of the configuration namespace. Successfully converts a configuration to a YAML-formatted string.

## Motivation and Context

Required to output simulation input parameters as part of the HDF in the standard HDF output from simulations.

## How Has This Been Tested?

Unit test added and fails

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have assigned/requested two reviewers for this pull request.

### CODE REVIEW

**Comment by andrewfullard** [comment]
Convert ConfigurationNameSpace to YAML and then dump to string instead?

File: tardis/io/util.py [hunk 1]
```diff
@@ -399,3 +399,75 @@ def download_from_url(url, dst):
                 pbar.update(1024)
     pbar.close()
     return file_size
+
+
```
Comment (wkerzendorf) [inline tardis/io/util.py:403]:
```suggestion

def quantity_representer(dumper, data):
    """
    docu goes here
    """
    return dumper.represent_data(str(data))
def cns_representer(dumper, data):
    """
    docu goes here
    """
    return dumper.represent_dict(dict(data))
```

**Comment by wkerzendorf** [comment]
`pd.Series({'config':x})`

File: tardis/io/tests/test_config_reader.py [hunk 2]
```diff
@@ -57,3 +58,13 @@ def test_from_config_dict(tardis_config_verysimple):
...
             tardis_config_verysimple, validate=True, config_dirname="test"
         )
+
+
+def test_config_hdf(hdf_file_path, tardis_config_verysimple):
+    expected = Configuration.from_config_dict(
+            tardis_config_verysimple, validate=True, config_dirname="test"
+        )
+    expected.to_hdf(hdf_file_path)
+    actual = pd.read_hdf(hdf_file_path, key="/config/config")
```
Comment (wkerzendorf) [inline tardis/io/tests/test_config_reader.py:68]: why config/config?

Comment (andrewfullard) [reply tardis/io/tests/test_config_reader.py]: Fixed in next commit

**Pull request merged**